test/framework/src/main/java/org/elasticsearch/cluster/service/FakeThreadPoolMasterService.java

```diff
@@ -157,7 +157,20 @@ public void onFailure(Exception e) {
                     scheduleNextTaskIfNecessary();
                 }
             }
-        }, wrapAckListener(ackListener));
+        };
+        threadPool.generic().execute(threadPool.getThreadContext().preserveContext(new Runnable() {
```

Comment:
NB not really the generic threadpool, we're using the `DeterministicTaskQueue` to run on a single thread here.